internal/util/kubernetes/kubernetes.go

```diff
@@ -67,6 +67,9 @@ func nestedNullableStringMap(obj map[string]interface{}, fields ...string) (map[
 
 // GetSecret gets secret using the kubernetes client
 func GetSecret(ctx context.Context, client k8sClient.Client, namespace, secretName string) (*corev1.Secret, error) {
+	if namespace == "" {
```

Comment:
nit: if we are going to check if namespace is empty then I guess we probably should also check if secretName is empty, right?